jina/peapods/zmq/__init__.py

```diff
@@ -59,12 +59,15 @@ def __init__(
         self.msg_recv = 0
         self.msg_sent = 0
         self.is_closed = False
+        self.is_closing = False
         self.opened_socks = []  # this must be here for `close()`
         self.ctx, self.in_sock, self.out_sock, self.ctrl_sock = self._init_sockets()
         self._register_pollin()
 
         self.opened_socks.extend([self.in_sock, self.out_sock, self.ctrl_sock])
         if self.in_sock_type == zmq.DEALER:
+            if self.args.dealer_startup_wait_time > 0:
```

Comment:
what is the rationale behind this waiting time?

Reply:
Currently, when starting all peas, the flow context uses the `wait_start_success` method to make sure all peas are started before sending the first data request.
Here, we close and start pods while data requests arrive. This means we can not ensure that the peas are ready when receiving their first request.
Therefore I made the wait_time configurable depending on the expected executor load time.
I think a better solution in the future would be to let the pea send the idle request after it is fully initialized.
So this is a workaround.

Reply:
The Runtime sends the idle request in its initialization right?

Reply:
Yes, the zmqlet of the pea. Not the flow context.